v3/mathjax3-ts/handlers/html/HTMLDocument.ts

```diff
@@ -0,0 +1,239 @@
+     */
+    public domStrings: HTMLDomStrings;
+
+    /*
+     * @override
+     * @constructor
+     * @extends{AbstractMathDocument}
+     */
+    constructor(document: any, options: OptionList) {
+        let [html, dom] = separateOptions(options, HTMLDomStrings.OPTIONS);
+        super(document, html);
+        this.domStrings = this.options['DomStrings'] || new HTMLDomStrings(dom);
+    }
+
+    /*
+     * Creates a Location object for a delimiter at the position given by index in the N's string
+     *  of the array of strings searched for math, recovering the origin DOM node where the delimiter
+     *  was found.
+     *
+     * @param{number} N             The index of the string in the string array
+     * @param{number} index         The position within the N's string that needs to be found
+     * @param{string} delim         The delimiter for this position
+     * @param{HTMLNodeArray} nodes  The list of node lists representing the string array
+     * @return{Location}            The Location object for the position of the delimiter in the document
+     */
+    protected findPosition(N: number, index: number, delim: string, nodes: HTMLNodeArray): Location {
+        for (const list of nodes[N]) {
+            let [node, n] = list;
+            if (index <= n) {
+                return {node: node, n: index, delim: delim};
+            } else {
+                index -= n;
+            }
+        }
+        return {node: null, n: 0, delim: delim};
+    }
+
+    /*
+     * Convert a ProtoItem to a MathItem (i.e., determine the actual Location
+     *  objects for its start and end)
+     *
+     * @param{ProtoItem} item       The proto math item to turn into an actual MathItem
+     * @param{InputJax} jax         The input jax to use for the MathItem
+     * @param{HTMLNodeArray} nodes  The array of node lists that produced the string array
+     * @return{HTMLMathItem}        The MathItem for the given proto item
+     */
+    protected mathItem(item: ProtoItem, jax: InputJax, nodes: HTMLNodeArray) {
+        let math = item.math;
+        let start = this.findPosition(item.n, item.start.n, item.open, nodes);
+        let end = this.findPosition(item.n, item.end.n, item.close, nodes);
+        return new HTMLMathItem(math, jax, item.display, start, end);
+    }
+
+    /*
+     * Get a list of containers (DOM nodes) to be searched for math.  These can be
+     *  specified by CSS selector, or as actual DOM elements or arrays of such.
+     *
+     * @param{(string | Element | Element[])[]} nodes  The array of items to make into a container list
+     * @param{Document} document                       The document in which to search
+     * @return{Element[]}                              The array of containers to search
+     */
+    protected getElements(nodes: (string | Element | Element[])[], document: Document) {
+        let containers: Element[] = [];
+        let NODELIST = document.childNodes.constructor as typeof NodeList;
```

Comment:
In node.js, there is no `window` or `document` object, so there is no global `NodeList` class to refer to (in a browser, `NodeList` would be on the `window` object).  Normally, you could look up the `window` from `document.defaultView`, but for a document created by `DOMParser.parseFromString()`, `document.defaultView` is null.  So I am getting the `NodeList` class from an instance of it (the `document.childNodes` array).

The reason we want this is we want to be able to process the results of calls like `document.selectorQueryAll('...')` without having to have the user convert to an array manually.  I really need to handle `HTMLCollection` as well, so will need to do more here anyway.  Let me see what I can do it improve the situation...